builder/remotecontext/git/gitutils.go

```diff
@@ -70,7 +70,7 @@ func fetchArgs(remoteURL *url.URL, ref string) []string {
 	shallow := true
 
 	if strings.HasPrefix(remoteURL.Scheme, "http") {
-		res, err := http.Head(fmt.Sprintf("%s/info/refs?service=git-upload-pack", remoteURL))
+		res, err := http.Get(fmt.Sprintf("%s/info/refs?service=git-upload-pack", remoteURL))
```

Comment:
Should call `res.Body.Close()` on successful request.

The `GET` request to get all remote refs is usually quite slow but it shouldn't matter here much as we are not consuming the body. I wouldn't mind actually having a whitelist with github here to avoid this request for that host.